### PR TITLE
[pallet-revive] implement the call data copy API 

### LINKED ISSUE #6770: Rework the input host functions to work like EVM
The `input` host function needs to be adapted to allow reading part of the input just like on EVM.

### PR DESCRIPTION
**Pull request by xermicus**
This PR implements the call data copy API by adjusting the input method.

Closes #6770

### CODE REVIEW

**Comment by xermicus** [comment]
bot fmt

**Review by athei — COMMENTED** [review]
Why are those PolkaVM files getting so much smaller? They were already 64bit before, right?

**Comment by xermicus** [comment]
> Why are those PolkaVM files getting so much smaller? They were already 64bit before, right?

Might be that the solc optimizer was still disabled if they were compiled with the remix backend.

File: substrate/frame/revive/uapi/src/host.rs [hunk 1]
```diff
@@ -245,18 +245,28 @@ pub trait HostFn: private::Sealed {
...
 	/// - the input was previously forwarded by a [`call()`][`Self::call()`].
+	/// - the `output` buffer is located in an PolkaVM invalid memory range.
 	///
 	/// # Parameters
 	///
-	/// - `output`: A reference to the output data buffer to write the input data.
-	fn input(output: &mut &mut [u8]);
+	/// - `output`: A reference to the output data buffer to write the call data.
+	/// - `offset`: The offset index into the call data from where to start copying.
+	fn call_data_copy(output: &mut &mut [u8], offset: u32);
```
[the remark this answers is not in the document]
Comment (xermicus) [reply substrate/frame/revive/uapi/src/host.rs:269]: Good catch thanks

**Review by athei — APPROVED** [review]

**Review by pgherveou — APPROVED** [review]

**Pull request merged**